File: exercise/java/day09/src/test/java/ClientTests.java
import account.Client;
import org.junit.jupiter.api.Test;

import java.util.LinkedHashMap;

import static java.lang.System.lineSeparator;
import static org.assertj.core.api.Assertions.assertThat;

class ClientTests {
    private final Client client = new Client(new LinkedHashMap<>() {{
        put("Tenet Deluxe Edition", 45.99);
        put("Inception", 30.50);
        put("The Dark Knight", 30.50);
        put("Interstellar", 23.98);
    }});

    @Test
    void client_should_return_statement() {
        String statement = client.toStatement();

        assertThat(client.getTotalAmount()).isEqualTo(130.97);
        assertThat(statement).isEqualTo(
                "Tenet Deluxe Edition for 45.99€" + lineSeparator() +
                        "Inception for 30.5€" + lineSeparator() +
                        "The Dark Knight for 30.5€" + lineSeparator() +
                        "Interstellar for 23.98€" + lineSeparator() +
                        "Total : 130.97€");
    }
}

File: exercise/kotlin/day09/src/test/kotlin/ClientTests.kt
import account.Client
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class ClientTests : StringSpec({
    val client = Client(
        mapOf(
            "Tenet Deluxe Edition" to 45.99,
            "Inception" to 30.50,
            "The Dark Knight" to 30.50,
            "Interstellar" to 23.98
        )
    )

    "client should return statement" {
        val statement = client.toStatement()
        client.totalAmount shouldBe 130.97
        statement shouldBe """|Tenet Deluxe Edition for 45.99€
                   |Inception for 30.5€
                   |The Dark Knight for 30.5€
                   |Interstellar for 23.98€
                   |Total : 130.97€""".trimMargin()
    }
})
